src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -51,6 +53,15 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
       stepToRender,
     } = options;
 
+    let { stateHandle } = currTransaction!.context;
+
+    if (step === 'cancel') {
+      authClient?.transactionManager.clear({ clearIdxResponse: false });
+      if (previousTransaction) {
+        stateHandle = previousTransaction.context.stateHandle;
+      }
```

Comment:
To explain why this fixes the issue: 

This is ultimately an auth-js issue, and we are adding in a temporary workaround to prevent broken links (this issue exists in v2 as well). 
[This](https://github.com/okta/okta-auth-js/blob/master/lib/idx/run.ts#L130) is the entrypoint in auth-js. If we follow the flow, you can see it eventually hits [this](https://github.com/okta/okta-auth-js/blob/master/lib/idx/introspect.ts#L30) line, which finally lands [here](https://github.com/okta/okta-auth-js/blob/master/lib/TransactionManager.ts#L340). This is problematic because it is expecting the stateHandle to match the stateHandle from the last request. However, we are sending the stateHandle from the current transaction (which was updated on the terminal response). When they do not match, this line returns null and causes `introspect` to be called instead of the `cancel` action. Calling introspect after a failure, just results in the same failure, causing this link to appear broken. The only way the user can escape this is by refreshing the page.

Reply:
Ticket opened to auth-js team: https://oktainc.atlassian.net/browse/OKTA-528448

Reply:
Attach a short explanation and ticket to the top of this block for context and also to discourage anyone from adding code inside of this logic for any reason